sdk/tables/azure-data-tables/tests/test_table_entity.py

```diff
@@ -898,23 +898,46 @@ def test_get_entity_if_match(self, tables_storage_account_name, tables_primary_s
         try:
             entity, etag = self._insert_random_entity()
 
-            # Act
-            # Do a get and confirm the etag is parsed correctly by using it
-            # as a condition to delete.
-            resp = self.table.get_entity(partition_key=entity['PartitionKey'],
-                                         row_key=entity['RowKey'])
+            entity = self.table.get_entity(
+                partition_key=entity['PartitionKey'],
+                row_key=entity['RowKey']
+            )
 
             self.table.delete_entity(
-                {"PartitionKey": resp['PartitionKey'], "RowKey": resp['RowKey']},
+                entity,
                 etag=etag,
                 match_condition=MatchConditions.IfNotModified
             )
 
             with pytest.raises(ResourceNotFoundError):
-                resp = self.table.get_entity(partition_key=entity['PartitionKey'],
-                                            row_key=entity['RowKey'])
+                self.table.get_entity(
+                    partition_key=entity['PartitionKey'],
+                    row_key=entity['RowKey']
+                )
+        finally:
+            self._tear_down()
+
+    @tables_decorator
+    def test_get_entity_if_match_entity_bad_etag(self, tables_storage_account_name, tables_primary_storage_account_key):
+        # Arrange
+        self._set_up(tables_storage_account_name, tables_primary_storage_account_key)
+        try:
+            entity, old_etag = self._insert_random_entity()
+
+            entity["value"] = 10
+            self.table.update_entity(entity)
+
+            # Get Entity and set old etag
+            e = self.table.get_entity(entity["PartitionKey"], entity["RowKey"])
+            new_etag = e.metadata["etag"]
+            e.metadata["etag"] = old_etag
+
+            with pytest.raises(HttpResponseError):
```

Comment:
Is this actually just a raw `HttpResponseError`? Or are we raising `azure.core.ResourceModifiedError`?

Reply:
This is a raw `HttpResponseError`, here's what it looks like.

```
E           azure.core.exceptions.HttpResponseError: The update condition specified in the request was not satisfied.
E           RequestId:0c9fe685-8002-003d-4cd4-41ac3f000000
E           Time:2021-05-05T17:33:24.9865841Z
E           ErrorCode:UpdateConditionNotSatisfied
```

Reply:
Interesting! I've made a note to fix that - but could come post-GA if needed.